examples/add/train_add_distill_sd_wds.py

```diff
@@ -1249,9 +1257,12 @@ def main(args):
     )
 
     # 3. Load text encoders from SD 1.X/2.X checkpoint.
-    text_encoder = CLIPTextModel.from_pretrained(
-        args.pretrained_teacher_model, subfolder="text_encoder", revision=args.teacher_revision
-    )
+    if args.use_pretrained_projection:
```

Comment:
Do we have any guidance on which one should be used here i.e., the projection model or the CLIPTextModel?

Reply:
From the [ADD paper](https://arxiv.org/pdf/2311.17042.pdf) it's not clear which is used. The ADD discriminator is based on the [StyleGAN-T](https://arxiv.org/pdf/2301.09515.pdf) discriminator, and the [StyleGAN-T code](https://github.com/autonomousvision/stylegan-t/blob/36ab80ce76237fefe03e65e9b3161c040ae888e3/networks/discriminator.py#L100-L102) suggests that they use a fixed length text embedding of shape `(batch_size, embed_dim)`. For CLIP models, this could be either the the `pooler_output` of `CLIPTextModel` (the class used by SD pipelines) or the `text_embeds` of `CLIPTextModelWithProjection`.

Reply:
I think it doesn't make much sense to use `pooler_output` for CLIP, it should  be`text_embeds`.